Sources/TuistKit/Utils/RootDirectoryLocator.swift

```diff
@@ -21,7 +21,7 @@ final class RootDirectoryLocator: RootDirectoryLocating {
     func locate(from path: AbsolutePath) -> AbsolutePath? {
         if let cachedDirectory = cached(path: path) {
             return cachedDirectory
-        } else if let tuistDirectory = FileHandler.shared.locateDirectory(Constants.tuistFolderName, traversingFrom: path) {
+        } else if let tuistDirectory = FileHandler.shared.locateDirectory(Constants.tuistDirectoryName, traversingFrom: path) {
```

Comment:
@ollieatkinson the issue you flagged does indeed occur (multiple Tuist directories). 

To mitigate this we'd need to handle the caching differently, the [`cached`](https://github.com/tuist/tuist/pull/542/files#diff-0c124dde2b83e5dd1ee0917ecfa27039R39) method would need to only return a cached path if one is present without traversing the cache.

i.e.

- Check `path` in cache
- Check the file system
- Repeat for the parent

Reply:
#630 should fix this